shared/clock/clock.cpp

```diff
@@ -17,12 +18,12 @@ namespace NUClear {
 
         // now is multiplied by the real time factor to sync
         // NUClear with the simulation time
-        auto now = NUClear::base_clock::now();
-        utility::clock::last_update =
-            clock::time_point(utility::clock::last_update
-                              + std::chrono::duration_cast<std::chrono::steady_clock::duration>(
-                                  (now - utility::clock::last_update) * utility::clock::custom_rtf));
-        return utility::clock::last_update;
+        auto now   = NUClear::base_clock::now();
+        auto delta = std::chrono::duration_cast<std::chrono::steady_clock::duration>((now - utility::clock::last_update)
+                                                                                     * utility::clock::custom_rtf);
+        utility::clock::state       = clock::time_point(utility::clock::state + delta);
+        utility::clock::last_update = now;
+        return utility::clock::state;
```

Comment:
After doing a manual trace, I can confirm that the old formula was broken and the new formula works.